src/3rd_party_repos.c

```diff
@@ -0,0 +1,132 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "3rd_party_repos.h"
+#include "swupd.h"
+
+#include <errno.h>
+
+#ifdef THIRDPARTY
+
+static char *get_repo_config_path(void)
+{
+	return str_or_die("%s/%s/%s", globals.state_dir, "3rd_party", "repo.ini");
```

Comment:
wouldn't it be better to use "sys_path_join" here instead of "str_or_die"?

Reply:
As a general rule yes, but here we have 3 paths to join, so I'm using the str_or_die() to prevent running it twice. We need a sys_path_join() that receives a variable number of parameters.